utils/sample_qc.py

```diff
@@ -364,82 +364,139 @@ def infer_sex(
+    :param flat f_stat_XY_cutoff: Max f-stat for XY samples
+    :param int lower_SD_cutoff: Number of standard deviations to use when determining upper limit of sex chromosome ploidy cutoffs
+    :param int upper_SD_cutoff: Permissive number of standard deviations to use when determining lower ploidy limit and aneuploidy cutoffs
+    :param int Y_lower_bound: Number of standard deviations to use as the lower bound for Y, specifically to correct for somatic loss of Y
     :rtype: StructExpression
     """
+    def get_ploidy_cutoffs(ht: hl.Table, sex: str, f_stat_cutoff: float, lower_SD_cutoff: int, upper_SD_cutoff: int, Y_lower_bound: int) -> dict:
+        """
+        Gets chromosome X and Y ploidy cutoffs for XY and XX samples
+
+        :param Table ht: Table with samples and their sex chromosome ploidies
+        :param str sex: Sex for which to check X/Y ploidies. Assumes input is either XX or XY.
+        :param float f_stat_cutoff: f-stat cutoff value. Assumes XX values will be above cutoff and XY values will be below cutoff.
+        :param int lower_SD_cutoff: Number of standard deviations to use when determining upper limit of sex chromosome ploidy cutoffs
+        :param int upper_SD_cutoff: Permissive number of standard deviations to use when determining lower ploidy limit and aneuploidy cutoffs
+        :param int Y_lower_bound: Number of standard deviations to use as the lower bound for Y, specifically to correct for somatic loss of Y
+        :return: Dictionary of ploidy cutoffs; key: sex chrom, values: [lower cut, upper cut, outlier (aneuploidy) cut]
+        :rtype: dict
+        """
+        if sex == "XX":
+            ht = ht.filter(ht.f_stat < f_stat_cutoff)
+        else:
+            ht = ht.filter(ht.f_stat > f_stat_cutoff)
+
+        # Get mean/stdev for chrX/Y ploidies
+        X_stats = ht.aggregate(hl.agg.stats(ht.chrX_ploidy))
+        Y_stats = ht.aggregate(hl.agg.stats(ht.chrY_ploidy))
+
+        # Set cutoffs -- keep cutoffs permissive to account for loss of sex chromosomes
+        cutoffs = {}
+        cutoffs["X"] = [
+                        X_stats.mean - (upper_SD_cutoff * X_stats.stdev), X_stats.mean + (lower_SD_cutoff * X_stats.stdev), 
+                        X_stats.mean + (upper_SD_cutoff * X_stats.stdev)
+                        ]
+        cutoffs["Y"] = [
+                        abs(min((Y_lower_bound * Y_stats.stdev), Y_stats.mean - (upper_SD_cutoff * Y_stats.stdev))), 
+                        Y_stats.mean + (lower_SD_cutoff * Y_stats.stdev), 
+                        Y_stats.mean + (upper_SD_cutoff * Y_stats.stdev)
+                        ]
+        return cutoffs
+   
+ 
+    xx_cutoffs = get_ploidy_cutoffs(ht, "XX", f_stat_XX_cutoff, lower_SD_cutoff, upper_SD_cutoff, Y_lower_bound)
+    xy_cutoffs = get_ploidy_cutoffs(ht, "XY", f_stat_XY_cutoff, lower_SD_cutoff, upper_SD_cutoff, Y_lower_bound)
+
+    xx_x_ploidy_cutoffs = xx_cutoffs["X"][:2]
+    xxx_x_ploidy_cutoff = xx_cutoffs["X"][2]
```

Comment:
As far as  I can tell, this is never set. (only two values are set for `cutoffs['X']`in `get_ploidy_cutoffs`